frog/imports/ui/GraphEditor/TopPanel/GraphList.jsx

```diff
@@ -1,31 +1,245 @@
+import ModeEdit from 'material-ui-icons/ModeEdit';
+import TextField from 'material-ui/TextField';
+import Input, { InputAdornment } from 'material-ui/Input';
+import Tooltip from 'material-ui/Tooltip';
 import { withTracker } from 'meteor/react-meteor-data';
 
-import { DropdownButton, MenuItem } from 'react-bootstrap';
-
+import { ValidButton } from '../Validator';
 import { connect, store } from '../store';
-import { Graphs } from '../../../api/graphs';
+import { Graphs, renameGraph } from '../../../api/graphs';
+
+const styles = theme => ({
+  root: {
+    display: 'flex',
+    minWidth: 0,
+    flexGrow: 1,
+    whiteSpace: 'nowrap'
+  },
+  basic: {
+    backgroundColor: 'white'
+  },
+  button: {
+    marginTop: theme.spacing.unit / 2,
+    padding: 3,
+    width: 35
+  },
+  textField: {
+    marginTop: theme.spacing.unit,
+    backgroundColor: 'white',
+    paddingRight: 0,
+    width: 200
+  },
+  spacer: {
+    backgroundColor: 'gray',
+    width: 50,
+    height: 1
+  },
+  validButton: {
+    backgroundColor: 'gray',
+    width: 50
+  },
+  durationTextField: {
+    marginTop: theme.spacing.unit,
+    backgroundColor: 'white',
+    paddingRight: 0,
+    width: 65
+  }
+});
+
+@withStyles(styles)
+class GraphListMenu extends React.Component {
+  state = {
+    isEditingName: false,
+    isEditingDuration: false,
+    graphName: '',
+    duration: ''
+  };
+
+  constructor(props) {
+    super(props);
+    const { graphId } = this.props;
+    this.selectedGraph = Graphs.findOne({ _id: graphId });
+    this.state = {
+      name: this.selectedGraph ? this.selectedGraph.name : 'untitled',
+      duration: this.selectedGraph ? this.selectedGraph.duration : 30
+    };
+  }
+
+  handleNameChange = e => {
+    e.preventDefault();
+    this.setState({
+      name: e.target.value
+    });
+  };
+
+  handleMenuChange = e => {
+    const selectedIndex = e.target.options.selectedIndex;
+    const graphId = e.target.options[selectedIndex].getAttribute('data-key');
+    this.selectedGraph = Graphs.findOne({ _id: graphId });
+    store.setId(graphId);
+    this.setState({
+      name: this.selectedGraph.name,
+      duration: this.selectedGraph.duration
+    });
+  };
+
+  handleDurationChange = e => {
+    e.preventDefault();
+    this.setState({
+      duration: e.target.value
+    });
+  };
+
+  handleGraphNameSubmit = e => {
+    e.preventDefault();
+    if (this.state.isEditingName) {
+      renameGraph(this.selectedGraph._id, this.state.name);
+    }
+    this.setState({ isEditingName: !this.state.isEditingName });
+  };
+
+  handleGraphDurationSubmit = e => {
+    e.preventDefault();
+    if (this.state.isEditingDuration) {
+      store.changeDuration(parseInt(this.state.duration, 10));
+    }
+    this.setState({ isEditingDuration: !this.state.isEditingDuration });
+  };
+
+  render() {
+    const { classes } = this.props;
+
+    return (
+      <div className={classes.root}>
+        {this.renderGraphSubComponent()}
```

Comment:
Why doing like this instead of 

```
<div>
     <GraphSubComponent />
     ....
</div>
}....

const GraphSubComponent = () => 
    <form>
        ....
    </form>

```

Reply:
Next level code :D

Reply:
✅